pkg/obfuscator/mac_address.go

```diff
@@ -19,8 +29,15 @@ func (m *macAddressObfuscator) Path(s string) string {
 func (m *macAddressObfuscator) Contents(s string) string {
 	matches := m.regex.FindAllString(s, -1)
 	for _, match := range matches {
-		s = strings.Replace(s, match, StaticMacReplacement, -1)
-		m.ReplacementTracker.AddReplacement(match, StaticMacReplacement)
+		var replacement string
+		switch m.replacementType {
+		case schema.ObfuscateReplacementTypeStatic:
+			replacement = m.GenerateIfAbsent(match, match, m.obfsGenerator.generateStaticReplacement)
+		case schema.ObfuscateReplacementTypeConsistent:
+			replacement = m.GenerateIfAbsent(match, match, m.obfsGenerator.generateConsistentReplacement)
+		}
```

Comment:
the whole switch looks also very similar to the one in the ip obfuscator, do you think we can refactor that too? I guess it would look the same for the domain one too.